crates/ssr/src/resolving.rs

```diff
@@ -214,24 +209,31 @@ impl<'db> ResolutionScope<'db> {
     }
 
     fn resolve_path(&self, path: &ast::Path) -> Option<hir::PathResolution> {
-        let hir_path = hir::Path::from_src(path.clone(), &self.hygiene)?;
         // First try resolving the whole path. This will work for things like
         // `std::collections::HashMap`, but will fail for things like
         // `std::collections::HashMap::new`.
-        if let Some(resolution) = self.scope.resolve_hir_path(&hir_path) {
+        if let Some(resolution) = self.scope.resolve_hypothetical(&path) {
             return Some(resolution);
         }
         // Resolution failed, try resolving the qualifier (e.g. `std::collections::HashMap` and if
         // that succeeds, then iterate through the candidates on the resolved type with the provided
         // name.
-        let resolved_qualifier = self.scope.resolve_hir_path_qualifier(&hir_path.qualifier()?)?;
+        let resolved_qualifier = self.scope.resolve_hypothetical(&path.qualifier()?)?;
         if let hir::PathResolution::Def(hir::ModuleDef::Adt(adt)) = resolved_qualifier {
+            let name = path.segment()?.name_ref()?;
             adt.ty(self.scope.db).iterate_path_candidates(
                 self.scope.db,
                 self.scope.module()?.krate(),
                 &self.scope.traits_in_scope(),
-                Some(hir_path.segments().last()?.name),
-                |_ty, assoc_item| Some(hir::PathResolution::AssocItem(assoc_item)),
+                None,
+                |_ty, assoc_item| {
+                    let item_name = assoc_item.name(self.scope.db)?;
+                    if item_name.to_string().as_str() == name.text().as_str() {
+                        Some(hir::PathResolution::AssocItem(assoc_item))
+                    } else {
+                        None
+                    }
+                },
```

Comment:
This bit is a little wonky, but I am not sure if we can improve it, while maintaining proper encapsulation... 

Long-term, names of AssocItems would be interrned&hygienic, so there wouldn't be a way for an IDE to conjure up a name out of thin air, and it would have to do approximate string matching.

Reply:
If we can't look up an AssocItem by name, then we wouldn't be able to then find references to it. Or are you saying we'd have to stop using find_usages?

Reply:
We can‘t look up by hygienic name in a hash table. We can find it by lossy string name via iteration.